internal/network/dual_stack_validations.go

```diff
@@ -0,0 +1,61 @@
+package network
+
+import (
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+// Verify if the constrains for dual-stack machine networks are met:
+//   * there are exactly two machine networks
+//   * the first one is IPv4 subnet
+//   * the second one is IPv6 subnet
+func VerifyMachineNetworksDualStack(networks []*models.MachineNetwork, isDualStack bool) error {
```

Comment:
Can you do some magic for these 3 duplicated functions?

Reply:
I was considering this, but my logic was as follows - today all the requirements are the same for those networks

* max 2 entries
* first one v4 
* and so on...

But those requirements in the upstream OCP are not aligned, e.g. the limit of clusterNetworks is something we impose as AI, but OCP does not. It should be relatively easy for us to support it, but we'd need to test it extensively so as for today, we don't. However, it already boils down to saying that in a very near future we will have different sets of requirements for different networks. As soon as we need to implement those, we'd need to undo the magic because the same function will not be able to handle different cases.

So, instead of this back-and-forth, I went with the simplest way that is - today we have 3 functions that look almost the same, but it enables us to already now modify validators independently.

What do you think ?

cc @flaper87 @celebdor

Reply:
I would also add that these 3 networks are not ruled by the same assumptions, either in OCP or in Assisted. Although it looks like that right now. The implementation would also be a bit hacky. 

I would go the python way here, explicit is better than implicit (and the type system is guiding us that way)